compiler/src/main/scala/org/scalajs/core/compiler/GenJSCode.scala

```diff
@@ -2955,9 +2960,10 @@ abstract class GenJSCode extends plugins.PluginComponent
           translateMatch(expr)
 
         // Sometimes the pattern matcher casts its final result
-        case Apply(TypeApply(Select(expr: LabelDef, nme.asInstanceOf_Ob), _), _)
+        case Apply(TypeApply(Select(expr: LabelDef, nme.asInstanceOf_Ob), targs), _)
```

Comment:
Could you replace `targs` by `List(targ)` just to be extra prudent? (maybe also the last `_` by `Nil`)?